### ISSUE TITLE
Image Color Picker for tags that need hex

### ISSUE THREAD

**Issue by ooosssay** [bryceco/GoMap#760]
Possibly add a image color picker by adding the plus sign next to tags that have a hex and when the user taps on it the screen that covers the imagery collapses then the user can select the place on the imagery what color they want then once they select it, the hex is inserted in to the field.

**Comment by bryceco** [comment]
That's an interesting idea. It only applies to rooftops where you can use satellite imagery, and I don't know how well satellite imagery reflects the actual color. And I would want to have some way to zoom in to pick the exact pixel(s) that are representative. If you're giving a color for an area (polygon) we should be able to calculate the color automatically, right?

**Comment by ooosssay** [comment]
> If you're giving a color for an area (polygon) we should be able to calculate the color automatically, right?

Yes, that's correct. If you have a polygon representing an area on a satellite image, you can calculate the average color automatically that is in that area. (Before this I was thinking about using multiple imagery’s and get the average color from that 😅 maybe that’s too much.)

Additionally, when doing this you could possibly enter in an existing box automatically (like a way) when the user presses the button. The user should also have the ability to enter in nodes onto the “wall” for when the object isn’t a square. And when the user finishes they can press a “Done” button. But when the user is selecting a color you could possibly add a banner saying something like “something (I don’t know) mode” and hide some ui elements like the settings, search, location and compass to signify to the user that he or she is currently getting color from the imagery and not editing the map.


> And I would want to have some way to zoom in to pick the exact pixel(s) that are representative.

What about something similar to the circle that is zoomed in when moving around a node?

> It only applies to rooftops where you can use satellite imagery, and I don't know how well satellite imagery reflects the actual color. 

Yes as half of an objects color can be affected by the sun but also a way we could solve the problem of having imagery not being color accurate is having a slider for both brightness and saturation but if I may add how about give the user 4 options being to open the camera and take a picture and use that to get a color or from an existing photo or from files (may be redundant) or from imagery.

I’m gonna be honest I was thinking something similar to iOS’s existing way of getting color from an image as it would match Swifts look.
